### PR TITLE
Adding timing capabilities to clean memory

### PR DESCRIPTION
**Pull request by plivesey**
This will enable projects to time how long garbage collection is taking

### CODE REVIEW

File: ConsistencyManager/ConsistencyManager.swift [hunk 1]
```diff
@@ -65,6 +65,22 @@ public class ConsistencyManager {
      */
     public var garbageCollectionInterval: NSTimeInterval = 300
 
+    // MARK: Constants
+
+    /**
+     This notification is fired whenever the asyncronous work of clean memory starts.
```
Comment (freshnitesh) [inline ConsistencyManager/ConsistencyManager.swift]: typo - asynchronous

File: ConsistencyManager/ConsistencyManager.swift [hunk 2]
```diff
@@ -65,6 +65,22 @@ public class ConsistencyManager {
...
+     It's useful if you want to add timers to this work.
+     Called on an internal thread.
+    */
+    public static let kCleanMemoryAsynchronousWorkStarted = "com.linkedin.consistencyManager.kCleanMemoryAsynchronousWorkStarted"
+
+    /**
+     This notification is fired whenever the asyncronous work of clean memory finishes.
```
Comment (freshnitesh) [inline ConsistencyManager/ConsistencyManager.swift]: typo - asynchronous

File: ConsistencyManagerTests/MemoryWarningTests.swift [hunk 3]
```diff
@@ -42,5 +54,10 @@ class MemoryWarningTests: ConsistencyManagerTestCase {
             XCTAssertEqual(listenersArray.count, 0)
         }
         // Else it's nil which is fine too
+
+        // Now, let's check we got the right start/finish times
+        XCTAssertEqual(cleanMemoryStartedTimes.count, 1)
+        XCTAssertEqual(cleanMemoryFinishedTimes.count, 1)
+        XCTAssertTrue(cleanMemoryStartedTimes[0].timeIntervalSince1970 <= cleanMemoryFinishedTimes[0].timeIntervalSince1970)
```
Comment (jmkk) [inline ConsistencyManagerTests/MemoryWarningTests.swift:63]: One more test case to add to make sure the notifications fire at expected times, register test start time, and at the end assert that cleanMemoryStartedTimes happened after that time.

Comment (plivesey) [inline ConsistencyManagerTests/MemoryWarningTests.swift:63]: Good idea.

Comment (plivesey) [inline ConsistencyManagerTests/MemoryWarningTests.swift:63]: Added

File: ConsistencyManagerTests/MemoryWarningTests.swift [hunk 4]
```diff
@@ -59,5 +61,6 @@ class MemoryWarningTests: ConsistencyManagerTestCase {
         XCTAssertEqual(cleanMemoryStartedTimes.count, 1)
         XCTAssertEqual(cleanMemoryFinishedTimes.count, 1)
         XCTAssertTrue(cleanMemoryStartedTimes[0].timeIntervalSince1970 <= cleanMemoryFinishedTimes[0].timeIntervalSince1970)
+        XCTAssertTrue(testStart.timeIntervalSinceNow <= cleanMemoryStartedTimes[0].timeIntervalSince1970)
```
Comment (jmkk) [inline ConsistencyManagerTests/MemoryWarningTests.swift]: Should be `testStart.timeIntervalSince1970`, not `timeIntervalSinceNow`

**Pull request merged**